app/lib/domain/controllers/language_controller.dart

```diff
@@ -0,0 +1,17 @@
+import 'package:word_link/domain/objects/cards_related/collection_object.dart';
+import 'package:word_link/domain/objects/cards_related/card_object.dart';
+
+part 'package:word_link/infrastructure/language_repository.dart';
+
+
+    abstract class LanguageController {
+    static LanguageController? _instance;
+
+    static LanguageController get instance => _instance ??= _LanguageRepository();
+
+    CollectionObject getMostUsedWords({required Language language, int numberOfWords = 5});
+
+    Map<String, String> _getWordMapForLanguage(Language language);
```

Comment:
There is no need to put a function in the controller if it is private.
Remove it from the controller